mkosi/config.py

```diff
@@ -1422,545 +1647,27 @@ def create_argument_parser(self) -> argparse.ArgumentParser:
-        )
-        group.add_argument(
-            "--hostname",
-            help="Set the system hostname",
-            metavar="HOSTNAME",
-            action=action,
-        )
-        group.add_argument(
-            "--root-password",
-            help="Set the system root password",
-            metavar="PASSWORD",
-            action=action,
-        )
-        group.add_argument(
-            "--root-shell",
-            help="Set the system root shell",
-            metavar="SHELL",
-            action=action,
-        )
+            long = s.long if s.long else f"--{s.dest.replace('_', '-')}"
+            opts = [s.short, long] if s.short else [long]
 
-        group = parser.add_argument_group("Validation options")
-        group.add_argument(
-            "--secure-boot",
-            metavar="BOOL",
-            help="Sign the resulting kernel/initrd image for UEFI SecureBoot",
-            nargs="?",
-            action=action,
-        )
-        group.add_argument(
-            "--secure-boot-key",
-            metavar="PATH",
-            help="UEFI SecureBoot private key in PEM format",
-            action=action,
-        )
-        group.add_argument(
-            "--secure-boot-certificate",
-            metavar="PATH",
-            help="UEFI SecureBoot certificate in X509 format",
-            action=action,
-        )
-        group.add_argument(
-            "--secure-boot-sign-tool",
-            metavar="TOOL",
-            help="Tool to use for signing PE binaries for secure boot",
-            action=action,
-        )
-        group.add_argument(
-            "--verity-key",
-            metavar="PATH",
-            help="Private key for signing verity signature in PEM format",
-            action=action,
-        )
-        group.add_argument(
-            "--verity-certificate",
-            metavar="PATH",
-            help="Certificate for signing verity signature in X509 format",
-            action=action,
-        )
-        group.add_argument(
-            "--sign-expected-pcr",
-            metavar="FEATURE",
-            help="Measure the components of the unified kernel image (UKI) and embed the PCR signature into the UKI",
-            action=action,
-        )
-        group.add_argument(
-            "--passphrase",
-            metavar="PATH",
-            help="Path to a file containing the passphrase to use when LUKS encryption is selected",
-            action=action,
-        )
-        group.add_argument(
-            "--checksum",
-            metavar="BOOL",
-            help="Write SHA256SUMS file",
-            nargs="?",
-            action=action,
-        )
-        group.add_argument(
-            "--sign",
-            help="Write and sign SHA256SUMS file",
-            metavar="BOOL",
-            nargs="?",
-            action=action,
-        )
-        group.add_argument("--key", help="GPG key to use for signing", action=action)
+            group.add_argument(    # type: ignore
```

Comment:
If you still have the warning in some log adding that to the commit message would be nice.